### PR TITLE
Protocol for updater (v2)

### PR DESCRIPTION
**Pull request by gabriel**
Instead of vendoring and using avdl, it's best to decouple. You'll notice there isn't much required in the new protocol.

There will be a later PR for changing to use the new protocol.

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers

**Comment by chrisnojima** [comment]
logically seems fine, need golang review

**Comment by maxtaco** [comment]
@jzila can you help with golang review?

File: protocol.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+// UpdatePromptResponse is the result for UpdatePrompt
+type UpdatePromptResponse struct {
+	Action     UpdateAction `json:"action"`
+	AutoUpdate bool         `json:"autoUpdate"`
+}
+
+// UpdatePromptUI is a prompt interface
+type UpdatePromptUI interface {
+	// UpdatePrompt prompts for an update
+	UpdatePrompt(Update, UpdateOptions, UpdatePromptOptions) (*UpdatePromptResponse, error)
```
Comment (jzila) [inline protocol.go]: why does this return a pointer type?

File: protocol.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+}
+
+// Time is milliseconds since epoch
+type Time int64
```
Comment (jzila) [inline protocol.go]: I don't think this type is helpful. `time.Unix()` takes an int64, and `time.Time.Unix()` returns an int64, so having a separate time here just adds casting complexity.

**Comment by jzila** [comment]
LGTM after nits

File: protocol.go (hunk 1 above)
Comment (gabriel) [inline protocol.go]: Wanted a nil response if the user didn't actually response (like on an error) instead of empty object.

Comment (jzila) [inline protocol.go]: The standard Go way of doing this is to return a zero value `UpdatePromptResponse{}`

File: protocol.go (hunk 2 above)
Comment (gabriel) [inline protocol.go]: Oh right... yeah I pulled this in from the avdl protocol side without thinking that I could nuke it instead... removed.

**Pull request merged**